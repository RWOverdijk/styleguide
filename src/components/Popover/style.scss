/** @define Popover */

@import "../../css/variables";
@import "../../../node_modules/bootstrap-sass/assets/stylesheets/bootstrap/variables";
@import "../../../node_modules/bootstrap-sass/assets/stylesheets/bootstrap/mixins";
@import "../../../node_modules/bootstrap-sass/assets/stylesheets/bootstrap/popovers";

.Van-Popover {
  font-family: "Proxima Nova", sans-serif;
  font-weight: $font-weight-light;
  line-height: initial;
  color: $color-primary;
  border-radius: 5px;
  box-shadow: 0 -5px 10px rgba(0, 0, 0, 0.2);

  &-trigger {
    cursor: pointer;
  }

  /* postcss-bem-linter: ignore */
  .popover-content {
    padding: 12px 16px;
    font-size: 14px;
  }
}
