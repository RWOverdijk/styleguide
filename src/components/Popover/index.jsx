import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';
import { OverlayTrigger, Popover as BootstrapPopover } from 'react-bootstrap';
import './style.scss';

class Popover extends PureComponent {
  getPopoverContent() {
    const { className, id, title, content, ...props } = this.props;

    return (
      <BootstrapPopover
        className={classnames('Van-Popover', className)}
        id={id}
        title={title}
        {...props}
      >
        {content}
      </BootstrapPopover>
    );
  }

  render() {
    const { rootClose, trigger, placement, children } = this.props;
    let triggerCopy = trigger;

    if (triggerCopy === 'hover') {
      triggerCopy = ['hover', 'focus'];
    }

    return (
      <OverlayTrigger
        trigger={triggerCopy}
        placement={placement}
        overlay={this.getPopoverContent()}
        rootClose={rootClose}
      >
        <span className="Van-Popover-trigger">{children}</span>
      </OverlayTrigger>
    );
  }
}

Popover.defaultProps = {
  id: '',
  className: '',
  title: '',
  content: '',
  children: '',
  placement: 'right',
  trigger: 'hover',
  rootClose: true
};

Popover.propTypes = {
  id: PropTypes.string,
  className: PropTypes.string,
  title: PropTypes.string,
  content: PropTypes.node,
  children: PropTypes.node,
  placement: PropTypes.oneOf(['top', 'right', 'bottom', 'left']),
  trigger: PropTypes.oneOf(['hover', 'focus', 'click']),
  rootClose: PropTypes.bool
};

export default Popover;
