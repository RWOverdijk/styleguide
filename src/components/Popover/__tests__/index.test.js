import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import Popover from '../';

const renderer = new ShallowRenderer();

describe("<Popover>", () => {
  it("renders Popover", () => {
    const tree = renderer.render(
      <Popover title="Popup title" content="String content">
        <span>Trigger content</span>
      </Popover>
    );

    expect(tree).toMatchSnapshot();
  });

  it("renders Popover on the left", () => {
    const tree = renderer.render(
      <Popover title="Popup title" content="String content" placement="left">
        <span>Trigger content</span>
      </Popover>
    );

    expect(tree).toMatchSnapshot();
  });

  it("renders Popover on the right", () => {
    const tree = renderer.render(
      <Popover title="Popup title" content="String content" placement="right">
        <span>Trigger content</span>
      </Popover>
    );

    expect(tree).toMatchSnapshot();
  });

  it("renders Popover on top", () => {
    const tree = renderer.render(
      <Popover title="Popup title" content="String content" placement="top">
        <span>Trigger content</span>
      </Popover>
    );

    expect(tree).toMatchSnapshot();
  });

  it("renders Popover on bottom", () => {
    const tree = renderer.render(
      <Popover title="Popup title" content="String content" placement="bottom">
        <span>Trigger content</span>
      </Popover>
    );

    expect(tree).toMatchSnapshot();
  });

  it("renders with focus as trigger if does not pass pass trigger", () => {
    const tree = renderer.render(
      <Popover title="Popup title" content="String content">
        <span>Trigger content</span>
      </Popover>
    );

    expect(tree).toMatchSnapshot();
  });

  it("renders with click as trigger", () => {
    const tree = renderer.render(
      <Popover title="Popup title" content="String content" trigger="click">
        <span>Trigger content</span>
      </Popover>
    );

    expect(tree).toMatchSnapshot();
  });

  it("renders with hover as trigger", () => {
    const tree = renderer.render(
      <Popover title="Popup title" content="String content" trigger="hover">
        <span>Trigger content</span>
      </Popover>
    );

    expect(tree).toMatchSnapshot();
  });

  it("renders with focus as trigger", () => {
    const tree = renderer.render(
      <Popover title="Popup title" content="String content" trigger="focus">
        <span>Trigger content</span>
      </Popover>
    );

    expect(tree).toMatchSnapshot();
  });

  it("renders with focus as trigger", () => {
    const content = (
      <div>
        <strong>asdasda</strong>
        <p>asdsad</p>
      </div>
    );
    const tree = renderer.render(
      <Popover title="Popup title" content={content}>
        <span>Trigger content</span>
      </Popover>
    );

    expect(tree).toMatchSnapshot();
  });

  it("renders with specific className", () => {
    const tree = renderer.render(
      <Popover title="Popup title" content="String content" className="my-class">
        <span>Trigger content</span>
      </Popover>
    );

    expect(tree).toMatchSnapshot();
  });

  it("renders with specific id", () => {
    const tree = renderer.render(
      <Popover title="Popup title" content="String content" id="my-id">
        <span>Trigger content</span>
      </Popover>
    );

    expect(tree).toMatchSnapshot();
  });

  it("renders with specific attributes", () => {
    const tree = renderer.render(
      <Popover title="Popup title" content="String content" data-id="0">
        <span>Trigger content</span>
      </Popover>
    );

    expect(tree).toMatchSnapshot();
  });
});
