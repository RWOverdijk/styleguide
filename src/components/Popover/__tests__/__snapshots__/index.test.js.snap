// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Popover> renders Popover 1`] = `
<OverlayTrigger
  defaultOverlayShown={false}
  overlay={
    <Popover
      bsClass="popover"
      className="Van-Popover"
      id=""
      placement="right"
      rootClose={true}
      title="Popup title"
      trigger="hover"
    >
      String content
    </Popover>
  }
  placement="right"
  rootClose={true}
  trigger={
    Array [
      "hover",
      "focus",
    ]
  }
>
  <span
    className="Van-Popover-trigger"
  >
    <span>
      Trigger content
    </span>
  </span>
</OverlayTrigger>
`;

exports[`<Popover> renders Popover on bottom 1`] = `
<OverlayTrigger
  defaultOverlayShown={false}
  overlay={
    <Popover
      bsClass="popover"
      className="Van-Popover"
      id=""
      placement="bottom"
      rootClose={true}
      title="Popup title"
      trigger="hover"
    >
      String content
    </Popover>
  }
  placement="bottom"
  rootClose={true}
  trigger={
    Array [
      "hover",
      "focus",
    ]
  }
>
  <span
    className="Van-Popover-trigger"
  >
    <span>
      Trigger content
    </span>
  </span>
</OverlayTrigger>
`;

exports[`<Popover> renders Popover on the left 1`] = `
<OverlayTrigger
  defaultOverlayShown={false}
  overlay={
    <Popover
      bsClass="popover"
      className="Van-Popover"
      id=""
      placement="left"
      rootClose={true}
      title="Popup title"
      trigger="hover"
    >
      String content
    </Popover>
  }
  placement="left"
  rootClose={true}
  trigger={
    Array [
      "hover",
      "focus",
    ]
  }
>
  <span
    className="Van-Popover-trigger"
  >
    <span>
      Trigger content
    </span>
  </span>
</OverlayTrigger>
`;

exports[`<Popover> renders Popover on the right 1`] = `
<OverlayTrigger
  defaultOverlayShown={false}
  overlay={
    <Popover
      bsClass="popover"
      className="Van-Popover"
      id=""
      placement="right"
      rootClose={true}
      title="Popup title"
      trigger="hover"
    >
      String content
    </Popover>
  }
  placement="right"
  rootClose={true}
  trigger={
    Array [
      "hover",
      "focus",
    ]
  }
>
  <span
    className="Van-Popover-trigger"
  >
    <span>
      Trigger content
    </span>
  </span>
</OverlayTrigger>
`;

exports[`<Popover> renders Popover on top 1`] = `
<OverlayTrigger
  defaultOverlayShown={false}
  overlay={
    <Popover
      bsClass="popover"
      className="Van-Popover"
      id=""
      placement="top"
      rootClose={true}
      title="Popup title"
      trigger="hover"
    >
      String content
    </Popover>
  }
  placement="top"
  rootClose={true}
  trigger={
    Array [
      "hover",
      "focus",
    ]
  }
>
  <span
    className="Van-Popover-trigger"
  >
    <span>
      Trigger content
    </span>
  </span>
</OverlayTrigger>
`;

exports[`<Popover> renders with click as trigger 1`] = `
<OverlayTrigger
  defaultOverlayShown={false}
  overlay={
    <Popover
      bsClass="popover"
      className="Van-Popover"
      id=""
      placement="right"
      rootClose={true}
      title="Popup title"
      trigger="click"
    >
      String content
    </Popover>
  }
  placement="right"
  rootClose={true}
  trigger="click"
>
  <span
    className="Van-Popover-trigger"
  >
    <span>
      Trigger content
    </span>
  </span>
</OverlayTrigger>
`;

exports[`<Popover> renders with focus as trigger 1`] = `
<OverlayTrigger
  defaultOverlayShown={false}
  overlay={
    <Popover
      bsClass="popover"
      className="Van-Popover"
      id=""
      placement="right"
      rootClose={true}
      title="Popup title"
      trigger="focus"
    >
      String content
    </Popover>
  }
  placement="right"
  rootClose={true}
  trigger="focus"
>
  <span
    className="Van-Popover-trigger"
  >
    <span>
      Trigger content
    </span>
  </span>
</OverlayTrigger>
`;

exports[`<Popover> renders with focus as trigger 2`] = `
<OverlayTrigger
  defaultOverlayShown={false}
  overlay={
    <Popover
      bsClass="popover"
      className="Van-Popover"
      id=""
      placement="right"
      rootClose={true}
      title="Popup title"
      trigger="hover"
    >
      <div>
        <strong>
          asdasda
        </strong>
        <p>
          asdsad
        </p>
      </div>
    </Popover>
  }
  placement="right"
  rootClose={true}
  trigger={
    Array [
      "hover",
      "focus",
    ]
  }
>
  <span
    className="Van-Popover-trigger"
  >
    <span>
      Trigger content
    </span>
  </span>
</OverlayTrigger>
`;

exports[`<Popover> renders with focus as trigger if does not pass pass trigger 1`] = `
<OverlayTrigger
  defaultOverlayShown={false}
  overlay={
    <Popover
      bsClass="popover"
      className="Van-Popover"
      id=""
      placement="right"
      rootClose={true}
      title="Popup title"
      trigger="hover"
    >
      String content
    </Popover>
  }
  placement="right"
  rootClose={true}
  trigger={
    Array [
      "hover",
      "focus",
    ]
  }
>
  <span
    className="Van-Popover-trigger"
  >
    <span>
      Trigger content
    </span>
  </span>
</OverlayTrigger>
`;

exports[`<Popover> renders with hover as trigger 1`] = `
<OverlayTrigger
  defaultOverlayShown={false}
  overlay={
    <Popover
      bsClass="popover"
      className="Van-Popover"
      id=""
      placement="right"
      rootClose={true}
      title="Popup title"
      trigger="hover"
    >
      String content
    </Popover>
  }
  placement="right"
  rootClose={true}
  trigger={
    Array [
      "hover",
      "focus",
    ]
  }
>
  <span
    className="Van-Popover-trigger"
  >
    <span>
      Trigger content
    </span>
  </span>
</OverlayTrigger>
`;

exports[`<Popover> renders with specific attributes 1`] = `
<OverlayTrigger
  defaultOverlayShown={false}
  overlay={
    <Popover
      bsClass="popover"
      className="Van-Popover"
      data-id="0"
      id=""
      placement="right"
      rootClose={true}
      title="Popup title"
      trigger="hover"
    >
      String content
    </Popover>
  }
  placement="right"
  rootClose={true}
  trigger={
    Array [
      "hover",
      "focus",
    ]
  }
>
  <span
    className="Van-Popover-trigger"
  >
    <span>
      Trigger content
    </span>
  </span>
</OverlayTrigger>
`;

exports[`<Popover> renders with specific className 1`] = `
<OverlayTrigger
  defaultOverlayShown={false}
  overlay={
    <Popover
      bsClass="popover"
      className="Van-Popover my-class"
      id=""
      placement="right"
      rootClose={true}
      title="Popup title"
      trigger="hover"
    >
      String content
    </Popover>
  }
  placement="right"
  rootClose={true}
  trigger={
    Array [
      "hover",
      "focus",
    ]
  }
>
  <span
    className="Van-Popover-trigger"
  >
    <span>
      Trigger content
    </span>
  </span>
</OverlayTrigger>
`;

exports[`<Popover> renders with specific id 1`] = `
<OverlayTrigger
  defaultOverlayShown={false}
  overlay={
    <Popover
      bsClass="popover"
      className="Van-Popover"
      id="my-id"
      placement="right"
      rootClose={true}
      title="Popup title"
      trigger="hover"
    >
      String content
    </Popover>
  }
  placement="right"
  rootClose={true}
  trigger={
    Array [
      "hover",
      "focus",
    ]
  }
>
  <span
    className="Van-Popover-trigger"
  >
    <span>
      Trigger content
    </span>
  </span>
</OverlayTrigger>
`;
